src/modules/job-manager/event.c

```diff
@@ -548,6 +601,47 @@ static int get_timestamp_now (double *timestamp)
     return 0;
 }
 
+int event_batch_process_event_entry (struct event *event,
+                                     struct job *job,
+                                     const char *name,
+                                     json_t *entry)
+{
+    struct events_listener *el;
+
+    if (event_batch_start (event) < 0)
+        goto error;
+
+    el = zlist_first (event->listeners);
+    while (el) {
+        if (el->filters) {
+            size_t index;
+            json_t *value;
+
+            json_array_foreach (el->filters, index, value) {
+                const char *str = json_string_value (value);
+                if (!strcmp (str, name)) {
+                    if (json_array_append (el->events, entry) < 0)
+                        goto nomem;
+                    event->batch->events_to_send = true;
+                }
+            }
```

Comment:
Perhaps if instead of a filters array, the request included optional allow or deny _objects_, then jansson's hashing could be utilized instead of walking the array for each element?  E.g.
```json
{
  "allow":{
    "priority":1,
    "foo":1
   }
}
```
The test then just becomes
```c
if ((el->allow && json_object_get (el->allow, name)) 
        || (el->deny && !json_object_get (el->deny, name)))
    json_array_append (el->events, entry)...
```

Reply:
Er that logic is slightly wrong since it doesn't cover if both allow and deny are missing, but my point wasn't to suggest allow/deny semantics - I _think_ that's a well undestood design pattern?